src/arrayMethodJoin.js

```diff
@@ -4,8 +4,27 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let str = '';
+    let connector = separator;
+
+    if (separator === null) {
+      connector = 'null';
+    }
+
+    for (let i = 0; i < this.length; i++) {
+      if (i !== this.length - 1 && this.length !== 0) {
+        if (this[i] === undefined || this[i] === null) {
+          str += connector;
+        } else {
+          str += this[i] + connector;
+        }
+      } else {
```

Comment:
This condition works if this.length = 0. Then how can you write str += this[i], if this[i] doesn't exist?